python/pyiceberg/io/pyarrow.py

```diff
@@ -786,15 +865,39 @@ def project_table(
     rows_counter = multiprocessing.Value("i", 0)
 
     with ThreadPool() as pool:
+        # Fetch the deletes
+        deletes_per_file: Dict[str, List[ChunkedArray]] = {}
```

Comment:
I think it would make more sense for this to be its own function. Something like `_read_all_delete_files`?